master/internal/stream/system.go

```diff
@@ -324,28 +368,42 @@ func doPublishLoop[T stream.Msg](
 			}
 			// Broadcast all the events.
 			publisher.Broadcast(events)
+			// XXX: look into why this is potentially an ineffective break statement
+			//nolint
 			break
```

Comment:
pretty sure it's because it is breaking out of the select/case statement, but it could confusingly appear as if it's breaking out of the `for` loop.

Prolly just delete this `break`, it's not doing anything.